build.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright © 2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::process::Command;
+
+fn main() {
+    let git_hash = Command::new("git")
+        .args(&["rev-parse", "HEAD"])
+        .output()
+        .expect("Expect to get git hash");
+
+    let git_status = Command::new("git")
+        .args(&["status", "--porcelain", "--untracked-files=no"])
+        .output()
+        .expect("Expect to get git status");
+
```

Comment:
What's wrong with `git describe` ?

Reply:
It will even tell you if the tree was dirty is dirty with `git describe --dirty`